lib/model/query/datasets.js

```diff
@@ -179,11 +179,18 @@ createPublishedDataset.audit.withResult = true;
 const _createPublishedProperty = (property) => sql`
 INSERT INTO ds_properties ("name", "datasetId", "publishedAt")
 VALUES (${property.name}, ${property.datasetId}, clock_timestamp())
+ON CONFLICT ("name", "datasetId")
+DO UPDATE SET "publishedAt" = clock_timestamp()
+WHERE ds_properties."publishedAt" IS NULL
```

Comment:
If two drafts would create the same property the property is still only in the database once, right?

Reply:
Yes, the property is only listed once, but there's another table `ds_property_fields` that links those properties to the forms/form_defs/specific form fields that they came from (if they came from a form and not the API).

With two draft forms, it'll look like this in the database.
```
ds_properties [
  { id: 1, name: 'first_name', datasetId: 1, publishedAt: null },
  { id: 2, name: 'age', datasetId: 1, publishedAt: null }
]
ds_property_fields [
  { dsPropertyId: 1, formDefId: 3, path: '/name' },
  { dsPropertyId: 2, formDefId: 3, path: '/age' },
  { dsPropertyId: 1, formDefId: 4, path: '/name' },
  { dsPropertyId: 2, formDefId: 4, path: '/age' }
]
```

Reply:
Great, makes sense. I was wondering whether there might be risk of the same property being published multiple times but it doesn’t seem so